### PR TITLE
#71 API to present reset password link to the user

### PR DESCRIPTION
**Pull request by Luciefifi**
# What does this PR do 
. A user should be presented with reset password link

# description of the task to be completed 
- designing UI
- request to reset email
- validate token
- reset user's password

# How should this be manually tested 
- clone the repo using git clone repo's URL
- change branch git checkout ft-reset-password
- create .env file and paste variables from .env.example
- run npm run dev
- visit the link 
- on the login page and visit the forgot password link 

# Relevant pivotal tracker #71

### CODE REVIEW

**Review by sostene12 — COMMENTED** [review]
running the branch, but nothing is being rendered

File: src/pages/ResetPassword.jsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import { useNavigate } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import { ToastContainer } from 'react-toastify';
+import Input from '../components/Auth/PasswordInput';
+import Button from '../components/Button/PasswordButton';
+import LoginIllustation from '../assets/login-illustration.svg';
+import { processReset } from '../redux/reducers/Auth/ResetPasswordSlice';
+import LoadingSpinner from '../components/LoadingSpinner';
+
+const override = css`
```
Comment (wakaflorien) [inline src/pages/ResetPassword.jsx]: - there is no need of using `@emotion/react` for styling, we have tailwind for that

File: src/pages/ResetPassword.jsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	const navigate = useNavigate();
+	const handleProcessReset = (event) => {
+		event.preventDefault();
+		dispatch(processReset({ token, password, confirmPassword }));
+	};
+
+	useEffect(() => {
+		let timer;
+		if (success) {
+			timer = setTimeout(() => navigate('/login'), 5000); // Redirect to login page after 5 seconds
```
Comment (wakaflorien) [inline src/pages/ResetPassword.jsx]: - it recommended to use `redirect` instead of     useNavigate` hook.

File: src/pages/ResetPassword.jsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+						type="password"
+						name="confirmPassword"
+						onChange={(event) => setConfirmPassword(event.target.value)}
+						autoComplete="new-password"
+					/>
+					{/* <div className="items-center">
+						{loading && (
+							<PacmanLoader color="#36D7B7" css={override} size={12} />
+						)}
+					</div> */}
```
Comment (wakaflorien) [inline src/pages/ResetPassword.jsx]: - use `className` prop to set input styles keeping in mind input width and height should be the same.
- check the values of left and right side division of the screen also.

File: src/pages/ResetPassword.jsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+					<ToastContainer />
+					<Button
```
Comment (wakaflorien) [inline src/pages/ResetPassword.jsx:73]: - check common size of button for uniformity.

File: src/pages/ForgotPassword.jsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+					className="w-592 left-35 h-592 top-216"
+				/>
+			</div>
+			<div className="w-5/6 flex flex-col h-full bg-brightGray   h-1024 p-9 left-346 py-40 ">
+				<h3 className="text-left font-extrabold text-2xl font-bold text-yellow py-2">
+					Request Reset
+				</h3>
+				<form onSubmit={handleRequestReset}>
+					<Input
+						placeholder="Email"
```
Comment (wakaflorien) [inline src/pages/ForgotPassword.jsx:47]: - check common size of the button.
- be consistent with the `padding` and `margins`

File: src/tests/resetPassword.test.jsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+	it('should dispay reset password button ', () => {
+		render(<ResetPassword />);
+		const button = screen.getByRole('button');
+		expect(button).toBeInTheDocument();
+	});
+
+	it('should render the ResetPassword component', () => {
+		render(<ResetPassword />);
+		const component = screen.getByRole('heading');
+		expect(component).toBeInTheDocument();
```
Comment (wakaflorien) [inline src/tests/resetPassword.test.jsx:30]: - Add more test scenarios other than component `render` and element `toBeInTheDocument`.

File: src/tests/forgotPassword.test.jsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+		expect(input).toBeInTheDocument();
+	});
+
+	it('should display a Request Reset button', () => {
+		render(<ForgotPassword />);
+		const button = screen.getByRole('button');
+		expect(button).toBeInTheDocument();
+	});
+
+	it('should display a "back to login" link', () => {
```
Comment (wakaflorien) [inline src/tests/forgotPassword.test.jsx:26]: - Add more test scenarios other than component render and element toBeInTheDocument.

**Review by wakaflorien — CHANGES_REQUESTED** [review]
- rename your branch according to the guide
- add coverage to `.gitignore`

File: coverage/api/index.html
Comment (sostene12) [inline coverage/api/index.html:1]: ignore coverage folder

File: coverage/pages/HomePage.jsx.html
Comment (sostene12) [inline coverage/pages/HomePage.jsx.html:1]: - please put coverage folder in gitignore, and don't push it

File: src/components/Button/PasswordButton.jsx
Comment (sostene12) [inline src/components/Button/PasswordButton.jsx:1]: change the styles of the button, remove the static styles and let all styles(classNames) be passed in props. 
```
<button type="submit" className={className}>
			{label}
		</button>
```

File: src/pages/ForgotPassword.jsx
Comment (sostene12) [inline src/pages/ForgotPassword.jsx:1]: please remove css, and use tailwind only

File: src/pages/index.jsx
Comment (sostene12) [inline src/pages/index.jsx:1]: may be remove that comments and delete this file use app file for routes

File: src/redux/reducers/Auth/ResetPasswordSlice.jsx
Comment (sostene12) [inline src/redux/reducers/Auth/ResetPasswordSlice.jsx:1]: ```http://localhost:3000/api/user/resetpassword/${token}` this is not a good practice, plase use the url from env 
`import varkeys from '../constants/keys';

const url = varkeys.APP_URL;
you can do like this  ${url}/api/user/resetpassword/${token}`
`

File: src/redux/reducers/auth/ForgotSlice.jsx
Comment (sostene12) [inline src/redux/reducers/auth/ForgotSlice.jsx:1]: follow the above comment , and remove that url of localhost and user the link from env, in addition use the deployed url

File: src/utilis/testUtilis.jsx
Comment (sostene12) [inline src/utilis/testUtilis.jsx:1]: i thinks this file is no longer necessary , hence this contents are in /test/setup.tests

File: src/tests/resetPassword.test.jsx
Comment (sostene12) [inline src/tests/resetPassword.test.jsx:1]: also change the file namings of those reducers

**Review by sostene12 — CHANGES_REQUESTED** [review]

File: src/components/Button/PasswordButton.jsx
Comment (sostene12) [inline src/components/Button/PasswordButton.jsx:1]: i suggest we use single button components, where the styles will be passed in the props

File: src/redux/reducers/rootreducer.js
Comment (sostene12) [inline src/redux/reducers/rootreducer.js:1]: you can changer the file extensions of the files in redux folder from .jsx to .js

**Review by sostene12 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**